src/backend/distributed/operations/stage_protocol.c

```diff
@@ -380,7 +380,13 @@ citus_update_table_statistics(PG_FUNCTION_ARGS)
 
 	Oid distributedTableId = PG_GETARG_OID(0);
 
-	UpdateTableStatistics(distributedTableId);
+	Relation relation = try_relation_open(distributedTableId, AccessShareLock);
+
+	if (relation != NULL)
+	{
+		UpdateTableStatistics(distributedTableId);
+		relation_close(relation, AccessShareLock);
+	}
```

Comment:
Should we keep this lock after the function is finished? That probably invites deadlocks.